src/query/plan/rewrite/index_lookup.hpp

```diff
@@ -534,6 +539,23 @@ class IndexLookupRewriter final : public HierarchicalLogicalOperatorVisitor {
       }
       return true;
     };
+
+    /*
+     * Comparator function between two indices. If new index has >= 10x vertices than the existing, it cannot be better.
+     * If it is <= 10x number of vertices, check average group size of property values. The index with smaller average
+     * group size is better. If the average group size is the same, choose the one closer to the uniform distribution.
+     * @param found: Current best label-property index.
+     * @param new_stats: Label-property index candidate.
+     * @param vertex_count: New index's number of vertices.
+     */
+    auto compare_indices = [](std::optional<LabelPropertyIndex> &found, std::optional<storage::IndexStats> &new_stats,
+                              int vertex_count) {
+      return new_stats.has_value() && found->index_stats.has_value() && (vertex_count / 10 <= found->vertex_count) &&
+             (utils::LessThanDecimal(new_stats->avg_group_size, found->index_stats->avg_group_size) ||
+              (utils::ApproxEqualDecimal(found->index_stats->avg_group_size, new_stats->avg_group_size) &&
+               utils::LessThanDecimal(new_stats->statistic, found->index_stats->statistic)));
+    };
```

Comment:
You did a good job as it is much cleaner now.

Few things that I noticed as I was reading description.

First, I thought this function does all the comparing due to description. But what it actually does is it compares part, and other part is below. I would say this is good approach, and to make it better we have few options.
One is to add all comparisons from below here. From below meaning:

```
if (!found || (vertex_count * 10 < found->vertex_count) || compare_indices(found, new_stats, vertex_count) ||
            found->vertex_count > vertex_count ||
            (found->vertex_count == vertex_count && is_better_type(filter.property_filter->type_))) {
          found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
}
```

Reason why is because function is named `compare_indices` and other parts of `if` in my opinion also compare indices. For example, why indices comparison is not `(vertex_count * 10 < found->vertex_count)` as it compares num vertices?

Then again there will be problem of multiple conditions and unreadability, and I would again opt for splitting accross multiple ifs:

```
if (!found)
    found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
else if (vertex_count * 10 < found->vertex_count) 
    found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
else if(...)
    found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
```

But this lambda seems to me like good approach, and I would just group it here.

Other option is to keep it without lambda and split it with multiple ifs.